docs/user-guide/connecting-applications.md

```diff
@@ -0,0 +1,205 @@
+```shell
+$ kubectl scale rc my-nginx --replicas=0; kubectl scale rc my-nginx --replicas=2;
+$ kubectl get pods -l app=nginx -o wide
+NAME             READY   STATUS     RESTARTS   AGE   NODE
+my-nginx-5j8ok   1/1     Running   	0         2m    node1
+my-nginx-90vaf   1/1     Running   0          2m    node2
+
+$ kubectl exec my-nginx-5j8ok -- printenv | grep SERVICE
+KUBERNETES_SERVICE_PORT=443
+NGINXSVC_SERVICE_HOST=10.0.116.146
+KUBERNETES_SERVICE_HOST=10.0.0.1
+NGINXSVC_SERVICE_PORT=80
+```
+
+### DNS:
+Kubernetes offers a DNS cluster addon Service that uses skydns to automatically assign dns names to other Services. You can check if it’s running on your cluster:
+```shell
+$ kubectl get services kube-dns --namespace=kube-system
+NAME       LABELS       SELECTOR             IP(S)       PORT(S)
+kube-dns   <none>       k8s-app=kube-dns     10.0.0.10   53/UDP
+                                                         53/TCP
+```
+If it isn’t running, you can [enable it](../../cluster/addons/dns/README.md#how-do-i-configure-it). The rest of this section will assume you have a Service with a long lived ip (nginxsvc), and a dns server that has assigned a name to that ip (the kube-dns cluster addon), so you can talk to the Service from any pod in your cluster using using standard methods (e.g. gethostbyname). Let’s create another pod to test this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox
+  namespace: default
+spec:
+  containers:
+  - image: busybox
+    command:
+      - sleep
+      - "3600"
+    imagePullPolicy: IfNotPresent
+    name: busybox
+  restartPolicy: Always
+```
+And perform a lookup of the nginx Service
+```shell
+$ kubectl get pods busybox
+NAME      READY     STATUS    RESTARTS   AGE
+busybox   1/1       Running   0          18s
+
+$ kubectl exec busybox -- nslookup nginxsvc
+Server:    10.0.0.10
+Address 1: 10.0.0.10
+Name:      nginxsvc
+Address 1: 10.0.116.146
+```
+
+## Exposing the Service to the internet
+
+For some parts of your applications you may want to expose a Service onto an external IP address. Kubernetes supports two ways of doing this: NodePorts and LoadBalancers.
+Lets recreate the nginx Service with a node port:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginxsvc
+spec:
+  type: NodePort
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+```
+
+If you have [firewall rules](../services-firewalls.md),  you will have to open the tcp port in the output of the kubectl command before you can access your Service:
```

Comment:
It works out of the box, but it doesn't look like we removed the warnings https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/create.go#L126